hydro_test/src/cluster/paxos.rs

```diff
@@ -483,7 +483,22 @@ fn recommit_after_leader_election<'a, P: PaxosPayload>(
     Stream<P2a<P>, Tick<Cluster<'a, Proposer>>, Bounded, NoOrder>,
     Optional<usize, Tick<Cluster<'a, Proposer>>, Bounded>,
 ) {
+    let p_p1b_max_checkpoint = accepted_logs.clone().fold_commutative(
```

Comment:
In theory you could replace this with a `.map().flatten().max()`.

Reply:
Doesn't this change behavior from a stream that contains either None or Some(checkpoint) to an Optional stream that may contain checkpoint? Because then the later join with `p_log_to_try_commit` won't go through if there is no checkpoint right? Without the flatten maybe it's ok?

Reply:
True! You can fix this by adding an addition `.into_singleton()` which should convert the `Optional<T>` into `Singleton<Option<T>>`.

Reply:
I'll just do `map().max()`, since Some(anything) > None so it works out